src/include/migraphx/process.hpp

```diff
@@ -49,9 +54,12 @@ struct MIGRAPHX_EXPORT process
     ~process() noexcept;
 
     process& cwd(const fs::path& p);
+    process& env(const std::vector<std::string>& envs);
+    process& launcher(const fs::path& launcher);
 
     void exec();
     void write(std::function<void(process::writer)> pipe_in);
+    void read(std::string& buffer) const;
```

Comment:
This should take `process::writer` so we can output directly to iostream without needing to go into a string buffer.